### PR TITLE
[DPE-1921] Handle delete_user exception

### PR DESCRIPTION
**Pull request by zmraul**
Addresses #104

- Change `current_acls` in `auth.py` to a property
- Added a method to iterate over relations to remove users. Not used, where to invoke? `update_status`?
- Change access to peer databag on `provider.py`

### CODE REVIEW

File: src/auth.py [hunk 1]
```diff
@@ -37,9 +39,14 @@ def __init__(self, charm):
...
         self.server_properties = self.charm.kafka_config.server_properties_filepath
 
-        self.current_acls: Set[Acl] = set()
         self.new_user_acls: Set[Acl] = set()
 
+    @property
+    def current_acls(self) -> Set[Acl]:
+        """Sets the current cluster ACLs."""
+        acls = self._get_acls_from_cluster()
+        return self._parse_acls(acls=acls)
```
Comment (marcoppenheimer) [inline src/auth.py:48]: **praise:** Nice!

File: src/auth.py [hunk 2]
```diff
@@ -196,7 +191,13 @@ def delete_user(self, username: str) -> None:
             f"--entity-name={username}",
             "--delete-config=SCRAM-SHA-512",
         ]
-        KafkaSnap.run_bin_command(bin_keyword="configs", bin_args=command)
+        try:
+            KafkaSnap.run_bin_command(bin_keyword="configs", bin_args=command)
+        except subprocess.CalledProcessError as e:
+            if "delete a user credential that does not exist" in e.stderr:
+                logger.info(f"User: {username} can't be deleted, it does not exist")
```
Comment (marcoppenheimer) [inline src/auth.py:198]: **nitpick/todo:** WARNING here

File: src/provider.py [hunk 3]
```diff
@@ -79,9 +78,9 @@ def on_topic_requested(self, event: TopicRequestedEvent):
             return
 
         # non-leader units need cluster_config_changed event to update their super.users
-        self.peer_relation.data[self.charm.app].update({username: password})
+        self.charm.app_peer_data.update({username: password})
 
-        self.kafka_auth.load_current_acls()
+        # self.kafka_auth.load_current_acls()
```
Comment (marcoppenheimer) [inline src/provider.py:83]: **nitpick:** comment can be removed?

File: src/provider.py [hunk 4]
```diff
@@ -79,9 +78,9 @@ def on_topic_requested(self, event: TopicRequestedEvent):
             return
 
         # non-leader units need cluster_config_changed event to update their super.users
-        self.peer_relation.data[self.charm.app].update({username: password})
+        self.charm.app_peer_data.update({username: password})
```
Comment (marcoppenheimer) [inline src/provider.py:81]: **praise:** Thanks for consolidating this!

**Review by marcoppenheimer — APPROVED** [review]
Really sensible changes, and really pleased with the 'leave the camp cleaner than you found it' approach :)

File: src/auth.py (hunk 1 above)
Comment (deusebio) [reply src/auth.py:48]: *praise* I really love using properties over mutable attribute. Well done!!!

**Review by deusebio — APPROVED** [review]
Sorry for holding off this review for some days, but things have been a bit busy lately

Tremendous work Raul! Well done! All sensible changes!

**Pull request merged**